pkg/controller/blockdeviceclaim/blockdeviceclaim_controller.go

```diff
@@ -354,9 +364,11 @@ func (r *ReconcileBlockDeviceClaim) getListofDevices(filters ...string) (*apis.B
 
 	opts := &client.ListOptions{}
 
-	// if filter strings are present, apply them before querying etcd
-	if len(filters) != 0 {
-		opts.SetLabelSelector(strings.Join(filters, ","))
+	if sel, err := v1.LabelSelectorAsSelector(selector); err != nil {
```

Comment:
In this case should we error out instead of continuing to list block devices without selector?

Reply:
Updated the code to error out. So if selector parsing fails, the complete listing and thus claiming will fail.